code/modules/mob/living/carbon/brain/MMI.dm

```diff
@@ -82,13 +88,73 @@ obj/item/device/mmi/Destroy()
 		src.loc = M//Should fix cybros run time erroring when blown up. It got deleted before, along with the frame.
 		M.mmi = src
 		return TRUE
+
+
+	if(istype(O,/obj/item/weapon/wrench))
+		var/builtempty = 0
+		for(var/t in sammi_assembly_parts)
+			var/cc=contents_count(t)
+			var/req=sammi_assembly_parts[t]
+			if(cc<req)
+				var/temppart = new t(src)
+				to_chat(user, "<span class='warning'>You're short [req-cc] [temppart]\s.</span>")
+				qdel(temppart)
+				temppart = null
+				return TRUE
+		if(!istype(loc,/turf))
+			to_chat(user, "<span class='warning'>You can't assemble the SAMMI, \the [src] has to be standing on the ground (or a table) to be perfectly precise.</span>")
+			return TRUE
+		if(!brainmob)
+			builtempty = 1
+			//to_chat(user, "<span class='warning'>What are you doing oh god put the brain back in.</span>")
+			//return TRUE
+		if(!builtempty)
+			if(!brainmob.key)
+				if(!mind_can_reenter(brainmob.mind))
+					to_chat(user, "<span class='notice'>\The [src] indicates that their mind is completely unresponsive; there's no point.</span>")
+					return TRUE
+			if(brainmob.stat == DEAD)
+				to_chat(user, "<span class='warning'>Yeah, good idea. Give something deader than the pizza in your fridge legs.  Mom would be so proud.</span>")
+				return TRUE
+			if(brainmob.mind in ticker.mode.head_revolutionaries)
+				to_chat(user, "<span class='warning'>The [src]'s firmware lets out a shrill sound, and flashes 'Abnormal Memory Engram'. It refuses to accept the brain.</span>")
```

Comment:
Yeah, actually, why IS this check here? If they 're dead, does it matter if their mind is a head rev?